File: TravellerCreator/ClientApp/src/app/sector-creator/models/rtt-worldgen/rtt-worldgen.star.ts
import {IStar} from "../interfaces/star";
import {Luminosity} from "../enums/luminosity";
import {RttWorldgenStarType} from "../enums/rttworldgen-star-type";
import {SpectralType} from "../enums/spectral-type";
import {CompanionOrbit} from "../enums/companion-orbit";

export class RttWorldgenStar implements IStar {
  id: string;
  rttWorldgenStarType: RttWorldgenStarType;
  spectralType: SpectralType;
  luminosity: Luminosity;
  spectralSubclass: number;
  companionOrbit: CompanionOrbit;
  expansionSize: number;
  age: number;
}

File: TravellerCreatorModels/SectorCreator/RTTWorldgen/RTTWorldgenStar.cs
using TravellerCreatorModels.SectorCreator.Enums;
using TravellerCreatorModels.SectorCreator.Interfaces;

namespace TravellerCreatorModels.SectorCreator.RTTWorldgen;

public class RTTWorldgenStar : IStar
{
    public Guid Id { get; } = Guid.NewGuid();
    public RTTWorldgenStarType RTTWorldgenStarType { get; set; }
    public SpectralType SpectralType { get; set; }
    public Luminosity Luminosity { get; set; }
    public int SpectralSubclass { get; set; }
    public CompanionOrbit CompanionOrbit { get; set; }
    public int ExpansionSize { get; set; }
    public int Age { get; set; }
}